File: samples/CS/14-ContextualTabs/RibbonItems.cs
using System;

namespace WinForms.Ribbon
{
    partial class RibbonItems
    {
        public void Init()
        {
            ButtonSelect.Click += new EventHandler<EventArgs>(_buttonSelect_ExecuteEvent);
            ButtonUnselect.Click += new EventHandler<EventArgs>(_buttonUnselect_ExecuteEvent);
        }

        void _buttonSelect_ExecuteEvent(object sender, EventArgs e)
        {
            if (TabGroupTableTools.ContextAvailable != ContextAvailability.Active)
                TabGroupTableTools.ContextAvailable = ContextAvailability.Active;
        }

        void _buttonUnselect_ExecuteEvent(object sender, EventArgs e)
        {
            if (TabGroupTableTools.ContextAvailable != ContextAvailability.NotAvailable)
                TabGroupTableTools.ContextAvailable = ContextAvailability.NotAvailable;
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/14-ContextualTabs/RibbonItems.vb
Imports System

Namespace WinForms.Ribbon

    Partial Class RibbonItems

        Public Sub Init()
            AddHandler _ButtonSelect.Click, AddressOf _buttonSelect_ExecuteEvent
            AddHandler _ButtonUnselect.Click, AddressOf _buttonUnselect_ExecuteEvent
        End Sub

        Private Sub _buttonSelect_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
            If _TabGroupTableTools.ContextAvailable <> ContextAvailability.Active Then
                _TabGroupTableTools.ContextAvailable = ContextAvailability.Active
            End If
        End Sub

        Private Sub _buttonUnselect_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
            If _TabGroupTableTools.ContextAvailable <> ContextAvailability.NotAvailable Then
                _TabGroupTableTools.ContextAvailable = ContextAvailability.NotAvailable
            End If
        End Sub

    End Class
End Namespace
